odbc/src/api/functions.rs

```diff
@@ -1521,7 +1523,13 @@ pub unsafe extern "C" fn SQLGetConnectAttrW(
                 let attributes = &conn.attributes.read().unwrap();
 
                 match attribute {
-                    ConnectionAttribute::CurrentCatalog => {
+                    // LoginTimeout
+                    103 => {
```

Comment:
Let's say that we have this enum in api/definitions (same as the one in odbc_sys, but it has FromPrimitive.
```#[repr(i32)]
#[derive(Debug, PartialEq, Eq, Clone, Copy, FromPrimitive)]
pub enum ConnectionAttribute {
    AsyncEnable = 4,
    AccessMode = 101,
    AutoCommit = 102,
    LoginTimeout = 103,
    Trace = 104,
    TraceFile = 105,
    TranslateLib = 106,
    TranslateOption = 107,
    TxnIsolation = 108,
    CurrentCatalog = 109,
    OdbcCursors = 110,
    QuietMode = 111,
    PacketSize = 112,
    ConnectionTimeout = 113,
    DisconnectBehaviour = 114,
    AsyncDbcFunctionsEnable = 117,
    AsyncDbcEvent = 119,
    EnlistInDtc = 1207,
    EnlistInXa = 1208,
    ConnectionDead = 1209,
    AutoIpd = 10001,
    MetadataId = 10014,
}
```

Then you can do this
```
match FromPrimitive::from_i32(attribute) {
     Some(valid_attr) => {
           ConnectionAttribute::CurrentCatalog => {...}
           ConnectionAttribute::LoginTimeout => {...}
           ...
           _ => {HYC00 - Optional feature not implemented}
     None => {HY092 - Invalid attribute/option identifier}
}
```

Reply:
Ok this makes a lot of sense! I'll hold off until our 1:1 later today to talk it out more to make sure we're definitely on the same page.

Reply:
I love this approach